### PR TITLE
Scribe: Final Updates

### PR DESCRIPTION
**Pull request by nautsimon**
**Description**
- livefill at tip
- testing livefill and multiple levels
- backend restructure
- indexer/fetcher structure
- scribe/node rework
- adding queries

Status
https://github.com/synapsecns/sanguine/commit/5cb85dca085aae80f5d2bc04c570306ab1ba0845 currently in prod

### CODE REVIEW

**Comment by trajan0x** [comment]
@nautsimon  I've submitted #1178 which adds support for a metrics endpoint on scribe and otelgql on scribe.

I'm gonna avoid merging this to avoid merge conflicts, but the substantive changes are:

image


and 


image


Feel free to merge my pr as well

File: ethergo/backends/mocks/simulated_test_backend.go [hunk 1]
```diff
@@ -296,13 +296,13 @@ func (_m *SimulatedTestBackend) ClientID() string {
 	return r0
 }
 
-// CodeAt provides a mock function with given fields: ctx, contract, blockNumber
-func (_m *SimulatedTestBackend) CodeAt(ctx context.Context, contract common.Address, blockNumber *big.Int) ([]byte, error) {
```
Comment (nautsimon) [inline ethergo/backends/mocks/simulated_test_backend.go:300]: Generate discrepancy

File: ethergo/chain/mocks/chain.go [hunk 2]
```diff
@@ -263,13 +263,13 @@ func (_m *Chain) ClientID() string {
 	return r0
 }
 
-// CodeAt provides a mock function with given fields: ctx, contract, blockNumber
```
Comment (nautsimon) [inline ethergo/chain/mocks/chain.go:266]: gen

File: ethergo/backends/mocks/simulated_test_backend.go [hunk 3]
```diff
@@ -296,13 +296,13 @@ func (_m *SimulatedTestBackend) ClientID() string {
...
-	ret := _m.Called(ctx, contract, blockNumber)
+// CodeAt provides a mock function with given fields: ctx, account, blockNumber
+func (_m *SimulatedTestBackend) CodeAt(ctx context.Context, account common.Address, blockNumber *big.Int) ([]byte, error) {
+	ret := _m.Called(ctx, account, blockNumber)
```
Comment (nautsimon) [inline ethergo/backends/mocks/simulated_test_backend.go:301]: Generate discrepancy.

**Pull request merged**